src/Features/CSharp/Portable/ExtractMethod/CSharpMethodExtractor.CSharpCodeGenerator.cs

```diff
@@ -86,9 +84,18 @@ private abstract partial class CSharpCodeGenerator : CodeGenerator<StatementSynt
             private CSharpSelectionResult CSharpSelectionResult
                 => (CSharpSelectionResult)SelectionResult;
 
-            protected override OperationStatus<IMethodSymbol> GenerateMethodDefinition(bool localFunction, CancellationToken cancellationToken)
+            public override OperationStatus<ImmutableArray<SyntaxNode>> GetNewMethodStatements(SyntaxNode insertionPointNode, CancellationToken cancellationToken)
+            {
+                var statements = CreateMethodBody(insertionPointNode, cancellationToken);
+                var status = CheckActiveStatements(statements);
+                return status.With(statements.CastArray<SyntaxNode>());
+            }
```

Comment:
code refactoring has to still make the new statements, to do some final applicability checks.

we migth be able to optimize this if it's still expensive to do less here in the codepath where we're just querying for applicability.